core/src/main/java/org/apache/iceberg/rest/requests/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * A REST request to create a namespace, with an optional set of properties.
```

Comment:
I think we can have the javadoc here for every request saying it maps to the XXXX in the OpenAPI spec, for more clarity of the readers who encounter this class. (I don't know if we can link that spec, if so it would be better)

Reply:
Earnest question: what should we say it maps to? The name? The name is the same as the classes in the OpenAPI spec. Is there something else we should link to?

As for actually linking, I agree that would be better though I'm not sure if it's doable. I'll look into it, but I suspect that it's not.

Reply:
I opened an issue to follow up on this. Since the class names currently match up with the class names in the spec, I think we can handle that as a follow up (I genuinely need to see in which ways is that possible - though maybe somebody else knows better already).

https://github.com/apache/iceberg/issues/4068

Reply:
Ohhh... like you mean `This maps to 'operationId: createNamespace'`? I'd like to avoid something that can drift, so if we can find sommething that we can specifically point to it (even if that's building the OpenAPI doc via code annotation generation in the future), I'd prefer that.

Reply:
I'll move this conversation over to the ticket 👍